include/rpm/rpmkeyring.h

```diff
@@ -107,6 +117,22 @@ rpmPubkey rpmPubkeyLink(rpmPubkey key);
  */
 char * rpmPubkeyBase64(rpmPubkey key);
 
+/** \ingroup rpmkeyring
+ * Return fingerprint of primary key
+ * @param key		Pubkey
+ * @param fp		Fingerprint data
+ * @param fplen		Length of Fingerprint
+ * @return		0 on success
+ */
+int rpmPubkeyFingerprint(rpmPubkey key, uint8_t **fp, size_t *fplen);
```

Comment:
Something that occurred to me just yesterday evening: do we actually need the binary version of the key fingerprint for anything? Dealing with a binary buffer is annoying in C, and doubly so if/when the caller only ever wants to print it in hex, which I think is the primary usecase for this.

So even if we want to retain the binary version, we should offer one that returns the hex string so that each and every caller doesn't need to juggle with that by themselves.

Reply:
Just added and instantly put to good use.